roles/pulp_api/tasks/generate_database_fields_key.yml

```diff
@@ -0,0 +1,47 @@
+---
+- name: Check for existing Pulp Database Encryption Key
+  stat:
+    path: '{{ __pulp_db_fields_key_path }}'
+    get_attributes: false
+    get_checksum: false
+    get_mime: false
+  register: __pulp_db_fields_key_file
+
+# A theoretically better approach would be to use the jinja2 filter
+# `b64encode` and not have to install `openssl`.
+# However, then the bytes to be encoded would have to be put
+# into a temporary ansible variable on the ansible management node, and the
+# templated file may be placed elsewhere on disk temporarily. It seems
+# more secure to just do it this way.
+
+# Fortunately every single distro we support uses this package name
+# for /usr/bin/openssl
+# https://pkgs.org/search/?q=%2Fusr%2Fbin%2Fopenssl
+- name: Install the openssl command
+  package:
+    name: openssl
+    state: present
+  become: true
+  when: not __pulp_db_fields_key_file.stat.exists
+
+- name: Generate Pulp Database Encryption Key
+  shell: >-
+    set -o pipefail &&
+    umask 0077 &&
+    dd if=/dev/urandom bs=32 count=1 2>/dev/null
```

Comment:
Apologies, I meant to comment on this before this PR got merged but we should probably switch this to use `openssl rand -base64 32`.